### ISSUE TITLE
the `@abstract` attribute seems to be redundant

### ISSUE THREAD

**Issue by yegor256** [objectionary/eo#3529]
We have a function `eo:abstract` and the `@abstract` attribute. It seems that the attribute is redundant: we can get rid of it and only live with the function.

**Comment by yegor256** [comment]
@maxonfjvipon WDYT?

**Comment by maxonfjvipon** [comment]
@yegor256 the function checks the presence of `@abstract` attribute. I'm not sure we can remove the attribute because there's no unique combination of other mandatory attributes like `@base` or `@name` which may replace `@abstract`

**Comment by maxonfjvipon** [comment]
@yegor256 or no. Actually if object does not have `@base` - it's already abstract. It seems that it's enough. WDYT?

**Comment by yegor256** [comment]
@maxonfjvipon yes, seems to be correct: no `@base` means "being abstract" or "being an atom". Maybe we can just simplify this to:

```
<xsl:sequence select="not(exists($o/@base)) and (exists($o/o) or $o/@atom)"/>
```

**Comment by maxonfjvipon** [comment]
@yegor256 even `exists($o/o)` is not necessary because the object can be empty

**Comment by yegor256** [comment]
@maxonfjvipon let's get rid of it!

**Comment by maxonfjvipon** [comment]
@yegor256 the problem appeared: void attributes also don't have `@base` and have only `@name` attribute. 
```
[x] > a
  [] > y
```
Here at the level of XMIR there's no difference between `x` and `y`, they both looks the same:
```xml
<o line="6" name="x" pos="1"/>
<o line="8" name="y" pos="2"/>
```

**Comment by maxonfjvipon** [comment]
@yegor256 yes, `@abstract` means formation, but void attribute is not formation

**Comment by yegor256** [comment]
@maxonfjvipon maybe let's do it this way (more compliant with the calculus):

```
<o line="6" name="x" pos="1" base="∅"/>
<o line="8" name="y" pos="2"/>
```

**Comment by maxonfjvipon** [comment]
@yegor256 this may work

**Comment by 0pdd** [comment]
@yegor256 2 puzzles [#3600](https://github.com/objectionary/eo/issues/3600), [#3601](https://github.com/objectionary/eo/issues/3601) are still not solved.

**Comment by yegor256** [comment]
@rultor release, tag is `0.46.0`

**Comment by 0pdd** [comment]
@yegor256 2 puzzles [#3600](https://github.com/objectionary/eo/issues/3600), [#4046](https://github.com/objectionary/eo/issues/4046) are still not solved; solved: [#3601](https://github.com/objectionary/eo/issues/3601).

**Comment by 0pdd** [comment]
@yegor256 all 3 puzzles are solved here: [#3600](https://github.com/objectionary/eo/issues/3600), [#3601](https://github.com/objectionary/eo/issues/3601), [#4046](https://github.com/objectionary/eo/issues/4046).
